projects/onto-search/src/lib/onto-search-facet/datepicker-facet/datepicker-facet.component.ts

```diff
@@ -0,0 +1,182 @@
+    this.data.selectedRange = this.getSearchDateFacetRange();
+  }
+
+  private mapFacetDates(facetGroupData: SearchDateFacetModel[]): void {
+    for (const facet of facetGroupData) {
+      const monthDate = moment(facet.facetData).startOf('month').toISOString();
+      const yearDate = moment(facet.facetData).startOf('year').toISOString();
+      this.daysMap.set(moment(facet.facetData).toISOString(), +facet.count);
+      const existingMonth = this.monthsMap.get(monthDate) || 0;
+      this.monthsMap.set(monthDate, existingMonth + +facet.count);
+      const existingYear = this.yearsMap.get(yearDate) || 0;
+      this.yearsMap.set(yearDate, existingYear + +facet.count);
+    }
+  }
+
+  private getSearchDateFacetRange(): SearchDateFacetRange {
+    const start: Moment = this.dateRangeGroup.get('start').value;
+    const end: Moment = this.dateRangeGroup.get('end').value;
+
+    return {
+      dateRange: {
+        start: start.toDate(),
+        end: end.toDate()
+      } as DateRange<Date>,
+      count: this.getCountForRange(start, end)
+    };
+  }
+
+  private getCountForRange(start: Moment, end: Moment): number {
+    const dayKeys = [...this.daysMap.keys()];
+    return dayKeys.filter((day) => start.isSameOrBefore(day) && end.isSameOrAfter(day))
+        .map((key) => this.daysMap.get(key))
+        .reduce((accumulator, count) => accumulator + count);
+  }
+
+  private addTooltipsToCalendarCells(view: string, datesMap: Map<string, number>, dateFormat: string): void {
+    const calendarCells = this.getCalendarCellsForView(view);
+    calendarCells.forEach((calendarCell) => {
+      let found = false;
+      datesMap.forEach((count, dateString) => {
+        const calendarCellDateString = moment(calendarCell.getAttribute('aria-label'), dateFormat);
+
+        if (calendarCellDateString.isSame(dateString)) {
+          found = true;
+          this.createTooltipForCalendarCell(calendarCell, count);
+        }
+      });
+
+      if (!found) {
+        this.removeTooltipFromCalendarCell(calendarCell);
+      }
+    });
+  }
+
+  private getCalendarCellsForView(view: string): any {
+    const elements = document.querySelectorAll('.onto-date-range-picker ' + view);
+    return elements.length > 0 ? elements[0].querySelectorAll('.mat-calendar-body-cell') : [];
+  }
+
+  private createTooltipForCalendarCell(calendarCell, tooltipText: number): void {
+    const div = calendarCell.querySelectorAll('div')[0];
+    div.classList.add('date-range-facet-tooltip');
+    const spans = calendarCell.querySelectorAll('span');
+    if (spans.length > 0) {
+      const span = spans[0];
+      span.innerHTML = tooltipText;
+    } else {
+      const span = document.createElement('span');
+      span.innerHTML = String(tooltipText);
+      span.classList.add('date-range-facet-tooltiptext');
+      div.appendChild(span);
+    }
+  }
+
+  private removeTooltipFromCalendarCell(calendarCell): void {
+    const div = calendarCell.querySelectorAll('div')[0];
```

Comment:
eventually add some methods for getting these `div` and `span` elements and reuse